src/com/owncloud/android/lib/resources/files/UploadRemoteFileOperation.java

```diff
@@ -63,20 +66,24 @@
 	protected static final String OC_TOTAL_LENGTH_HEADER = "OC-Total-Length";
 	protected static final String IF_MATCH_HEADER = "If-Match";
 
-	protected String mLocalPath;
 	protected String mRemotePath;
 	protected String mMimeType;
 	protected PutMethod mPutMethod = null;
 	protected boolean mForbiddenCharsInServer = false;
 	protected String mRequiredEtag = null;
+	protected FileInputStream mInputStream;
 	
 	protected final AtomicBoolean mCancellationRequested = new AtomicBoolean(false);
 	protected Set<OnDatatransferProgressListener> mDataTransferListeners = new HashSet<OnDatatransferProgressListener>();
 
 	protected RequestEntity mEntity = null;
 
 	public UploadRemoteFileOperation(String localPath, String remotePath, String mimeType) {
-		mLocalPath = localPath;
+		try {
+			mInputStream = new FileInputStream(new File(localPath));
```

Comment:
This is making fail the automatic tests; before the operation didn't fail until was run, now it's failing in construction.

Let's delay the decision on this just a bit, since there is a chance that we need to roll back this change fully; content:// URIs, in a general sense, are not apropriate to be stored.